src/common/io/http/response.c

```diff
@@ -0,0 +1,412 @@
+
+/***********************************************************************************************************************************
+Object type
+***********************************************************************************************************************************/
+struct HttpResponse
+{
+    MemContext *memContext;                                         // Mem context
+
+    HttpSession *session;                                           // HTTP session
+    IoRead *contentRead;                                            // Read interface for response content
+
+    unsigned int code;                                              // Response code (e.g. 200, 404)
+    String *reason;                                                 // Response reason e.g. (OK, Not Found)
+    HttpHeader *header;                                             // Response headers
+
+    bool contentChunked;                                            // Is the response content chunked?
+    uint64_t contentSize;                                           // Content size (ignored for chunked)
+    uint64_t contentRemaining;                                      // Content remaining (per chunk if chunked)
+    bool closeOnContentEof;                                         // Will server close after content is sent?
+    bool contentExists;                                             // Does content exist?
+    bool contentEof;                                                // Has all content been read?
+    Buffer *content;                                                // Caches content once requested
+};
+
+OBJECT_DEFINE_MOVE(HTTP_RESPONSE);
+OBJECT_DEFINE_FREE(HTTP_RESPONSE);
+
+OBJECT_DEFINE_GET(IoRead, , HTTP_RESPONSE, IoRead *, contentRead);
+OBJECT_DEFINE_GET(Code, const, HTTP_RESPONSE, unsigned int, code);
+OBJECT_DEFINE_GET(Header, const, HTTP_RESPONSE, const HttpHeader *, header);
+OBJECT_DEFINE_GET(Reason, const, HTTP_RESPONSE, const String *, reason);
+
+/***********************************************************************************************************************************
+When response is done close/reuse the connection
+***********************************************************************************************************************************/
+static void
+httpResponseDone(HttpResponse *this)
+{
+    FUNCTION_LOG_BEGIN(logLevelTrace);
+        FUNCTION_LOG_PARAM(HTTP_RESPONSE, this);
+    FUNCTION_LOG_END();
+
+    ASSERT(this != NULL);
+    ASSERT(this->session != NULL);
+
+    // If close was requested by the server then free the session
+    if (this->closeOnContentEof)
+    {
+        httpSessionFree(this->session);
+
+        // Only update the close stats after a successful response so it is not counted if there was an error/retry
+        httpClientStat.close++;
+    }
+    // Else return it to the client do it can be reused
```

Comment:
do -> so   (Else return it to the client **so** it can be reused)

Reply:
Fixed.